framework/src/auxkernels/NearestNodeValueAux.C

```diff
@@ -35,12 +35,15 @@ NearestNodeValueAux::NearestNodeValueAux(const InputParameters & parameters)
     _paired_variable(coupled("paired_variable"))
 {
   if (boundaryNames().size() > 1)
-    mooseError("NearestNodeValueAux can only be used with one boundary at a time!");
+    paramError("boundary", "NearestNodeValueAux can only be used with one boundary at a time!");
+  if (boundaryNames().empty())
+    paramError("boundary", "Should be specified on a boundary or group of boundaries");
```

Comment:
If we're erroring if there's more than one boundary, seems like we should remove the second half of this sentence
```suggestion
    paramError("boundary", "Should be specified on a boundary");
```